### PR TITLE
migrated to rxjava2

### PR DESCRIPTION
**Pull request by BugsBunnyBR**
I know it is a lot of changes and if you prefer I can change the target branch of this PR, just create a rx2 branch.

### CODE REVIEW

**Comment by pwittchen** [comment]
Hi,

Thanks for the PR and your effort! It's a lot of work! :-) I don't want to abandon support for RxJava1.x in this library. I'd prefer to keep RxJava1.x version alongside RxJava2.x in a separate module or Git branch (I don't know, which approach is better). For now, I'd prefer to stick to RxJava1.x on the master branch. I've created RxJava2.x branch and I think it would be a better idea to change target branch to RxJava2.x in your PR. I haven't reviewed your code yet, but I'll do it soon.

Regards,
Piotr

File: library/build.gradle [hunk 1]
```diff
@@ -2,38 +2,37 @@ apply plugin: 'com.android.library'
...
-  }
 }
 
 dependencies {
-  compile 'io.reactivex:rxjava:1.2.1'
-  androidTestCompile 'com.android.support.test:testing-support-lib:0.1'
-  androidTestCompile('com.google.truth:truth:0.28') {
-    exclude group: 'junit' // Android has JUnit built in
-  }
+    compile 'io.reactivex.rxjava2:rxjava:2.0.7'
```
Comment (pwittchen) [inline library/build.gradle]: You're putting different versions of RxJava2.x in several modules. In app-kotlin module you provided RxJava v. 2.0.1. Please, make all dependencies consistent

Comment (BugsBunnyBR) [inline library/build.gradle]: RxJava (2.0.7) is in a different version of RxAndroid (2.0.1), that is what is different.
https://github.com/ReactiveX/RxAndroid

**Comment by BugsBunnyBR** [comment]
I changed the target branch. :-D

File: library/build.gradle (hunk 1 above)
Comment (pwittchen) [inline library/build.gradle]: Ok, Sorry. I've missed it.

**Comment by pwittchen** [comment]
Great! We're almost good to go! I'll test it locally and hopefully merge it soon. :)
Thanks!

**Comment by BugsBunnyBR** [comment]
After the merge of this PR... would you release a new version of the library? Maybe changing the package ?

**Comment by pwittchen** [comment]
I'm planning to change `POM_ARTIFACT_ID` in `library/gradle.properties` file on RxJava2.x branch from `reactivebeacons` to `reactivebeacons-rx2`.
After that, anyone who would like to use this library with RxJava2.x will need to set or update appropriate dependency in `build.gradle` file. It will be still possible to use RxJava1.x and eventually update the library with older dependency in the future.

**Comment by pwittchen** [comment]
Changes are merged. I also needed to update code responsible for runtime permissions. I'm going to organize a bit gradle config before future release.

**Pull request merged**